packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -97,7 +97,7 @@ function resolveSync(
     const dir = path.dirname(name);
     let result;
     if (isDirectory(dir)) {
-      result = resolveAsFile(name) || resolveAsDirectory(name);
+      result = resolveAsDirectory(name) || resolveAsFile(name);
```

Comment:
idk why we do `resolveAsFile` at all if we're inside `isDirectory`...

Reply:
this probably isn't the correct fix, based on looking at the `resolve` implementation: https://github.com/browserify/resolve/blob/bd3f55188f0f3d1dac1e29c2f9a0eaa0878d4bd0/lib/sync.js#L69

`resolve` does not have the same error Jest exhibits here, so either they've fixed it since Jest forked, or something was lost when forking and the subsequent refactoring